packages/aws-cdk-lib/aws-logs/lib/transformer.ts

```diff
@@ -0,0 +1,1393 @@
+}
+
+/**
+ * This object defines one key that will be moved with the moveKey processor.
+ */
+export interface MoveKeyEntryProperty {
+  /**
+   * The key to move.
+   */
+  readonly source: string;
+
+  /**
+   * The key to move to.
+   */
+  readonly target: string;
+  /**
+   * Specifies whether to overwrite the value if the target key already exists.
+   * @default false
+   */
+  readonly overwriteIfExists?: boolean;
+}
+
+/**
+ * This object defines one key that will be renamed with the renameKey processor.
+ */
+export interface RenameKeyEntryProperty {
+  /**
+   * The key to rename.
+   */
+  readonly key: string;
+
+  /**
+   * The string to use for the new key name.
+   */
+  readonly renameTo: string;
+
+  /**
+   * Whether to overwrite the target key if it already exists.
+   * @default false
+   */
+  readonly overwriteIfExists?: boolean;
+}
+
+/**
+ * This object defines one log field that will be split with the splitString processor.
+ */
+export interface SplitStringEntryProperty {
+  /**
+   * The key of the field to split.
+   */
+  readonly source: string;
+
+  /** The separator character to split the string on */
+  readonly delimiter: DelimiterCharacter;
+}
+
+/**
+ * This object defines one log field key that will be replaced using the substituteString processor.
+ */
+export interface SubstituteStringEntryProperty {
+  /**
+   * The key to modify.
+   */
+  readonly source: string;
+
+  /**
+   * The regular expression string to be replaced.
+   */
+  readonly from: string;
+
+  /**
+   * The string to be substituted for each match of from.
+   */
+  readonly to: string;
+}
+
+/**
+ * This object defines one value type that will be converted using the typeConverter processor.
+ */
+export interface TypeConverterEntryProperty {
+  /**
+   * The key with the value that is to be converted to a different type.
+   */
+  readonly key: string;
+
+  /** The data type to convert the field value to. */
+  readonly type: TypeConverterType;
+}
+
+/**
+ * The Resource properties for AWS::Logs::Transformer resource. This
+ * interface defines all configuration options for the CfnTransformer construct.
+ */
+export interface TransformerAttributes {
+  /**
+   * Name of the transformer.
+   */
+  readonly transformerName: string;
+  /** Existing log group that you want to associate with this transformer. */
+  readonly logGroup: ILogGroup;
+  /** List of processors in a transformer */
+  readonly transformerConfig: Array<IProcessor>;
+}
+
+/**
+ * The Resource properties for AWS::Logs::Transformer resource. This
+ * interface defines all configuration options for the CfnTransformer construct.
+ */
+export interface TransformerProps {
+  /**
+   * Name of the transformer.
+   */
+  readonly transformerName: string;
+  /** Existing log group that you want to associate with this transformer. */
+  readonly logGroup: ILogGroup;
+  /** List of processors in a transformer */
+  readonly transformerConfig: Array<IProcessor>;
```

Comment:
Why do we call this `transformerConfig`? `processors` seems to be more natural name?